src/repository.ts

```diff
@@ -81,14 +85,14 @@ export class StorageRepository implements Repository {
     return undefined;
   }
 
-  getSegment(identifier: string, cacheable = true): Segment {
+  async getSegment(identifier: string, cacheable = true): Promise<Segment> {
     const segmentKey = this.formatSegmentKey(identifier);
     let segment = this.cache.get(segmentKey) as Segment;
     if (segment) {
       return segment;
     }
     if (this.store) {
-      segment = this.store.get(segmentKey) as Segment;
+      segment = await this.store.get<Segment>(segmentKey);
```

Comment:
Definitely not something to think about now, but at some point we need to go through all of the SDK's and replace `Segment` with `TargetGroup` to make sure we have consistency across UI and SDK.

Reply:
make sense